README.md

```diff
@@ -73,18 +73,71 @@ Such an object might need a description as well, so you can also annotate it:
 }
 ```
 
-Again, the naming rule `#` joined with the fields name must be followed, so the `docsonnet` utility can automatically join together the contents of your object with it's annotated description.
+Again, the naming rule `#` joined with the fields name must be followed, so the `docsonnet` utility can automatically join together the contents of your object with its annotated description.
 
-## FAQ
 
-#### Do my projects need to have `doc-util` installed to vendor/?
+## Usage
+
+Once you have a Jsonnet library annotated with `doc-util`, you can generate the docs using one of three ways:
+
+- [Jsonnet renderer](#jsonnet-renderer)
+- [docsonnet binary](#docsonnet-binary)
+- [docsonnet docker image](#docsonnet-docker-image)
+
+### Jsonnet renderer
+
+The docs can be rendered using Jsonnet with the
+[render](https://github.com/jsonnet-libs/docsonnet/tree/master/doc-util#fn-render) function.
+
+In your library source, add a file `docs.jsonnet` (assuming your library entrypoint is `main.libsonnet`) with the
+following contents:
+
+```jsonnet
+local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
+d.render(import 'main.libsonnet')
+```
+
+Then, you can render the markdown to the `docs/` folder using the following command:
+
+```
+jsonnet -S -c -m docs/ docs.jsonnet
+```
+
+Note that this requires `doc-util` to be installed to `vender/` to work properly.
+
+### docsonnet binary
+
+Alternatively, the docs can be rendered using the `docsonnet` go binary. The `docsonnet` binary embeds the `doc-util`
+library, avoiding the need to have `doc-util` installed.
 
-No! The `docsonnet` binary comes included with it, and during normal Jsonnet use the docsonnet keys will never be accessed, so your Jsonnet runs just fine without.
+You can install the `docsonnet` binary using `go install`:
 
-> **Note** 
-> 
+```
+go install github.com/jsonnet-libs/docsonnet@master
+```
+
+Once the binary is installed, you can generate the docs by passing it the main entrypoint to your Jsonnet library:
+
+```
+docsonnet main.libsonnet
+```
+
+> **Note**
+>
 > Linters like [jsonnet-lint](https://pkg.go.dev/github.com/google/go-jsonnet/linter) or `tk lint` require the imports to be resolvable, so you should add `doc-util` to `vendor/` when using these linters.
 
+### docsonnet docker image
+
+You can also use the [docker image](https://hub.docker.com/r/jsonnetlibs/docsonnet) which contains the `docsonnet`
+binary if you do not wish to set up go or install the binary locally:
+
+```
+docker run -v "$(pwd):/src" jsonnetlibs/docsonnet /src/main.libsonnet
```

Comment:
docker run --rm -v "$(pwd):/src" -v "$(pwd)/docs:/docs" jsonnetlibs/docsonnet /src/main.libsonnet

with the original command I had no output, therefore the second -v
also added --rm because I'm not sure it's worth keeping those  instances

One thing, it creates the doc as root user

Reply:
Ah good catch! Updated to include the second `-v`: [d9c0dc9](https://github.com/jsonnet-libs/docsonnet/pull/38/commits/d9c0dc9717b24a4c198f0bf264878c5a2c357756)